### PR TITLE
admission/priority: externalize priority admission controller

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

**Work in Progress** - This PR may include not needed changes, but I'm currently in experimenting phase. Update is up to follow.

This PR addresses portion of #66680, by externalizing Priority Admission controller.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Addresses portion of #66680.

**Release note**:
```release-note
NONE
```

/assign @deads2k @yue9944882 @sttts

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold work-in-progress

Currently experimenting with something, those changes may not be needed and will be discussed once I find what's all needed.

**Comment by xmudrii** [comment]
/test pull-kubernetes-verify

verifying there is no flake

File: plugin/pkg/admission/priority/admission.go [hunk 1]
```diff
@@ -221,7 +222,7 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 // validatePriorityClass ensures that the value field is not larger than the highest user definable priority. If the GlobalDefault is set, it ensures that there is no other PriorityClass whose GlobalDefault is set.
 func (p *priorityPlugin) validatePriorityClass(a admission.Attributes) error {
 	operation := a.GetOperation()
-	pc, ok := a.GetObject().(*scheduling.PriorityClass)
+	pc, ok := a.GetObject().(*schedulingv1beta1.PriorityClass)
```
Comment (sttts) [inline plugin/pkg/admission/priority/admission.go]: the admission plugin still gets an internal type.

Background: admission is part of the server chain.

File: plugin/pkg/admission/priority/admission.go [hunk 2]
```diff
@@ -78,20 +79,20 @@ func (p *priorityPlugin) ValidateInitialization() error {
...
+func (p *priorityPlugin) SetExternalKubeInformerFactory(f informers.SharedInformerFactory) {
+	priorityInformer := f.Scheduling().V1beta1().PriorityClasses()
 	p.lister = priorityInformer.Lister()
 	p.SetReadyFunc(priorityInformer.Informer().HasSynced)
 }
 
 var (
 	podResource           = api.Resource("pods")
-	priorityClassResource = scheduling.Resource("priorityclasses")
+	priorityClassResource = schedulingv1beta1.Resource("priorityclasses")
```
Comment (sttts) [inline plugin/pkg/admission/priority/admission.go]: where is this used?

Comment (yue9944882) [reply plugin/pkg/admission/priority/admission.go]: L131. this change shouldn't happen is all the obj contained in the admission attributes should be internal

**Comment by wenjiaswe** [comment]
@yliaog

File: plugin/pkg/admission/priority/admission_test.go [hunk 3]
```diff
@@ -22,23 +22,28 @@ import (
...
 
 func addPriorityClasses(ctrl *priorityPlugin, priorityClasses []*scheduling.PriorityClass) {
 	informerFactory := informers.NewSharedInformerFactory(nil, controller.NoResyncPeriodFunc())
-	ctrl.SetInternalKubeInformerFactory(informerFactory)
+	ctrl.SetExternalKubeInformerFactory(informerFactory)
 	// First add the existing classes to the cache.
 	for _, c := range priorityClasses {
-		informerFactory.Scheduling().InternalVersion().PriorityClasses().Informer().GetStore().Add(c)
+		var s schedulingv1beta1.PriorityClass
+		runtime.Must(v1beta1.Convert_scheduling_PriorityClass_To_v1beta1_PriorityClass(c, &s, nil))
```
Comment (yue9944882) [inline plugin/pkg/admission/priority/admission_test.go]: please return an error instead of panic

Comment (xmudrii) [reply plugin/pkg/admission/priority/admission_test.go]: Fixed. Does it look okay now?

File: plugin/pkg/admission/priority/admission_test.go [hunk 4]
```diff
@@ -220,7 +231,10 @@ func TestDefaultPriority(t *testing.T) {
 	for _, test := range tests {
 		glog.V(4).Infof("starting test %q", test.name)
 		ctrl := newPlugin()
-		addPriorityClasses(ctrl, test.classesBefore)
+		err := addPriorityClasses(ctrl, test.classesBefore)
```
Comment (sttts) [inline plugin/pkg/admission/priority/admission_test.go]: nit: `if err := ...; err != nil {`

**Comment by sttts** [comment]
@deads2k approved?

File: plugin/pkg/admission/priority/admission_test.go (hunk 4 above)
Comment (xmudrii) [reply plugin/pkg/admission/priority/admission_test.go]: :+1:

**Comment by deads2k** [comment]
/approve

intentionally not setting the milestone. This is valuable, but not important enough to shove into 1.12 at the last moment.

**Pull request merged**